tests/BaseElementTest.php

```diff
@@ -236,4 +247,55 @@ public function testOnBeforeWriteNoParent()
 
         $this->assertEquals(0, (int) $element1->Sort);
     }
+
+    public function getElementCMSLinkDataProvider()
+    {
+        return [
+            'element1' => [
+                TestElement::class,
+                'elementDataObject1',
+                'http://localhost/test/1/ItemEditForm/field/ElementalArea/item/4/edit',
+                true
+            ],
+            'element2' => [
+                TestElementDataObject::class,
+                'testElementDataObject1',
+                'http://localhost/test/1/ItemEditForm/field/ElementalArea/item/5/edit',
+            ],
+            'element3' => [
+                ElementContent::class,
+                'contentDataObject1',
+                'http://localhost/test/1/ItemEditForm',
+            ],
+            'element4' => [
+                ElementContent::class,
+                'content1',
+                'http://localhost/admin/pages/edit/show/1',
+            ],
+            'element5' => [
+                ElementContent::class,
+                'contentDataObject2',
+                null
+            ],
+        ];
+    }
+
+    /**
+     * @param string $class
+     * @param string $element
+     * @param string|null $link
+     * @dataProvider getElementCMSLinkDataProvider
+     */
+
+    public function testGetElementCMSLink($class, $element, $link, $directLink = false)
```

Comment:
```suggestion
     * @dataProvider getElementCMSLinkDataProvider
     */
    public function testGetElementCMSLink(string $class, string $element, ?string $link, bool $directLink = false): void
```
Strongly type the method instead of adding phpdoc params.